src/rgw/rgw_op.cc

```diff
@@ -4104,6 +4104,10 @@ void RGWPostObj::execute()
     policy.encode(aclbl);
     emplace_attr(RGW_ATTR_ACL, std::move(aclbl));
 
+    bufferlist scbl;
+    scbl.append(s->info.storage_class.get_storage_class());
+    attrs[RGW_ATTR_STORAGE_CLASS] = scbl;
```

Comment:
@joke-lee see my newer code at `wip-rgw-tiering-4`, that has this commit:
https://github.com/yehudasa/ceph/commit/ef5968a27e1db6ad1b0ba7c685e2ec59417f17e2

Need to make sure that  `s->dest_placement` set correctly (maybe after `s->info.storage_class` is retrieved). Then will need to use it here.